server/datastore/mysql/scripts.go

```diff
@@ -549,7 +616,12 @@ ON DUPLICATE KEY UPDATE
 
 		// insert the new scripts and the ones that have changed
 		for _, s := range incomingScripts {
-			if _, err := tx.ExecContext(ctx, insertNewOrEditedScript, tmID, globalOrTeamID, s.Name, s.ScriptContents); err != nil {
+			scRes, err := insertScriptContents(ctx, s.ScriptContents, tx)
+			if err != nil {
+				return ctxerr.Wrapf(ctx, err, "inserting script contents for script with name %q", s.Name)
+			}
+			id, _ := scRes.LastInsertId()
```

Comment:
I'm not sure that would return the `id` if the script content already existed, right? Do we have a test case that verifies that? I think the `id` would need to be read using the `md5_checksum` as `WHERE` condition.